proxy_agent/src/proxy/proxy_listener.rs

```diff
@@ -35,14 +35,42 @@ pub fn start_async(port: u16, pool_size: u16, shared_state: Arc<Mutex<SharedStat
         });
 }
 
+/// start listener at the given address with retry logic
+fn start_listener(addr: &str) -> std::io::Result<TcpListener> {
+    for _ in 0..5 {
+        let listener = TcpListener::bind(addr);
+        match listener {
+            Ok(l) => {
+                return Ok(l);
+            }
+            Err(e) => match e.kind() {
+                std::io::ErrorKind::AddrInUse => {
+                    let message = format!("Failed bind to '{}' with error 'AddrInUse'.", addr);
+                    logger::write_error(message);
```

Comment:
do we need to add a sleep here before continue?

Reply:
yes, I cut it to somewhere else.